### PR TITLE
New version: ownCloud.ownCloudDesktop version 4.2.0.11670

### PR DESCRIPTION
**Pull request by SpecterShell**
- [X] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [X] This PR only modifies one (1) manifest
- [X] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [X] Have you tested your manifest locally with `winget install --manifest <path>`?
- [X] Does your manifest conform to the [1.5 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.5.0)?

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/120031)

### CODE REVIEW

**Comment by SpecterShell** [comment]
```
2023-09-16 22:10:43.175 [CLI ] Starting: 'C:\Users\VALIDA~1\AppData\Local\Temp\WinGet\ownCloud.ownCloudDesktop.4.2.0.11670\ownCloud-4.2.0.11670.x86.msi' with arguments '/passive /norestart /log "C:\Users\Validator\AppData\Local\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir\WinGet-ownCloud.ownCloudDesktop.4.2.0.11670-2023-09-16-22-10-43.174.log"'
2023-09-16 22:10:47.710 [CLI ] ShellExecute installer failed: 3010
2023-09-16 22:10:47.714 [CLI ] Terminating context: 0x8a150109 at D:\a\_work\1\s\external\pkg\src\AppInstallerCLICore\Workflows\InstallFlow.cpp:1bc
```
I think 3010 should also be considered as successful installation for validation pipeline.

File: manifests/o/ownCloud/ownCloudDesktop/4.2.0.11670/ownCloud.ownCloudDesktop.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+- Architecture: x86
+  InstallerUrl: https://download.owncloud.com/desktop/ownCloud/stable/4.2.0.11670/win/ownCloud-4.2.0.11670.x86.msi
+  InstallerSha256: C83BFBC874D75EE0768DAC72B88E8BCE036D1F6BD6AE7C066FB53B8477C74111
+  ProductCode: '{D1F3B372-1ED4-4DAF-B6E8-44B33A1BC53A}'
+  AppsAndFeaturesEntries:
+  - ProductCode: '{D1F3B372-1ED4-4DAF-B6E8-44B33A1BC53A}'
+    UpgradeCode: '{EB15081D-1892-4DB5-872C-9424BFFFF22F}'
+- Architecture: x64
+  InstallerUrl: https://download.owncloud.com/desktop/ownCloud/stable/4.2.0.11670/win/ownCloud-4.2.0.11670.x64.msi
+  InstallerSha256: D98695B19B11ACF0847C61B3214F68934E2BB087D6431EAD05E388DE154A1804
```
Comment (stephengillie) [inline manifests/o/ownCloud/ownCloudDesktop/4.2.0.11670/ownCloud.ownCloudDesktop.installer.yaml:22]:
```suggestion
  InstallerSha256: D98695B19B11ACF0847C61B3214F68934E2BB087D6431EAD05E388DE154A1804
  InstallerSuccessCodes:3010
```

Comment (stephengillie) [reply manifests/o/ownCloud/ownCloudDesktop/4.2.0.11670/ownCloud.ownCloudDesktop.installer.yaml:22]: I'm hoping that I put the correct syntax.

Comment (SpecterShell) [reply manifests/o/ownCloud/ownCloudDesktop/4.2.0.11670/ownCloud.ownCloudDesktop.installer.yaml:22]: https://learn.microsoft.com/en-us/windows/win32/msi/error-codes
3010 stands for "installation succeed and restart required" for MSI installers and WinGet will notify user a restart is required in this case. Doesn't this unnecessarily suppress that?

**Comment by stephengillie** [comment]
Hi @SpecterShell,

Yes, specifying `3010` as an InstallerSuccessCode might unnecessarily suppress the notice that a reboot is necessary. I was confused about your goals, since your immediately previous reply stated: 
> I think 3010 should also be considered as successful installation for validation pipeline.

I was trying to fulfill this request without considering the code itself. This package seems to be indicating that a reboot is required. You're right that we shouldn't block this code in this way. 

Please let me know how you would like to proceed.

**Review by stephengillie — APPROVED** [review]

**Pull request merged**